PurchasesCoreSwift/Identity/IdentityManager.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  IdentityManager.swift
+//
+//  Created by Joshua Liebowitz on 8/9/21.
+
+import Foundation
+
+// TODO (post-migration): Change back to internal, including all public properties.
+@objc(RCIdentityManager) public class IdentityManager: NSObject {
+
+    @objc public var currentAppUserID: String? {
+        return deviceCache.cachedAppUserID
+    }
+
+    @objc public var currentUserIsAnonymous: Bool {
+        let anonymousRegex = #"\$RCAnonymousID:([a-z0-9]{32})$"#
+        let anonymousFoundRange = currentAppUserID?.range(of: anonymousRegex, options: .regularExpression)
+        let foundAnonymous = anonymousFoundRange != nil && !(anonymousFoundRange?.isEmpty ?? false)
+        let currentAppUserIDLooksAnonymous = foundAnonymous
+
+        let isLegacyAnonymousAppUserID: Bool
+        if let cachedAppUserID = currentAppUserID {
+            isLegacyAnonymousAppUserID = cachedAppUserID == deviceCache.cachedLegacyAppUserID
+        } else {
+            isLegacyAnonymousAppUserID = false
+        }
+
+        return currentAppUserIDLooksAnonymous || isLegacyAnonymousAppUserID
+    }
+
+    private let deviceCache: DeviceCache
+    private let backend: Backend
+    private let purchaserInfoManager: PurchaserInfoManager
+
+    @objc public init(deviceCache: DeviceCache, backend: Backend, purchaserInfoManager: PurchaserInfoManager) {
+        self.deviceCache = deviceCache
+        self.backend = backend
+        self.purchaserInfoManager = purchaserInfoManager
+    }
+
+    @objc public func configure(appUserID maybeAppUserID: String?) {
+        let appUserID = maybeAppUserID
+            ?? deviceCache.cachedAppUserID
+            ?? deviceCache.cachedLegacyAppUserID
+            ?? generateRandomID()
+        Logger.user(String(format: Strings.identity.identifying_app_user_id, appUserID))
+
+        deviceCache.cache(appUserID: appUserID)
+        deviceCache.cleanupSubscriberAttributes()
```

Comment:
not really migration-related, but if this is called with the currentAppUserId, should we avoid calling anything?

Reply:
Huh, I'm not sure! If we do have the `currentAppUserID` maybe it makes sense to skip the whole thing. I'll add the logic and see what happens...

Reply:
So, here's what I end up with:
```swift
        if (maybeCurrentAppUserID != nil && maybeAppUserID != nil) && maybeAppUserID != maybeCurrentAppUserID {
            return
        }
```
It turns out, if both are nil, we still want to run the configure logic. However, if they are both non-nil, and they equal each other, then we should just skip this.

@aboedo, thoughts? (I'm pushing this change up)

Reply:
currently, `DeviceCache` [ensures that things are only deleted for the current user](https://github.com/RevenueCat/purchases-ios/blob/main/Purchases/Caching/RCDeviceCache.m#L364)
This behavior has been preserved in the migration, so we should be good. 

This method does 2 things, though: 
- deletes synced attributes for other users
- migrates from v1 subscriber attributes format to v2

So we should still have this happen for the current user

Reply:
Filed #732 and #731